src/components/Tabset/index.js

```diff
@@ -48,6 +86,34 @@ export default class Tabset extends Component {
         tabChildren[tabIndex].ref.focus();
     }
 
+    updateTabsetButtonsVisibility() {
+        const tabset = this.tabsetRef.current;
+        const tabsetScrollWidth = tabset.scrollWidth;
+        const tabsetScrollLeft = tabset.scrollLeft;
+        const screenWidth = window.innerWidth;
+        const tabsetPosition = tabset.offsetLeft;
+        const tabsetWidth = tabset.offsetWidth;
+        const tabsetMaxScroll = tabsetScrollWidth - tabsetWidth;
+        const tabsetChildren = tabset.children;
+        const tabsetChildrenArray = [].slice.call(tabsetChildren, 0);
+        const tabsetChildrenWidth = tabsetChildrenArray.map(child => child.offsetWidth);
+        const tabsetChildrenTotalWidth = tabsetChildrenWidth.reduce(
+            (prev, curr) => prev + curr,
+            );
+        this.setState({ tabsetPosition });
+        this.setState({ screenWidth });
+        this.setState({ tabsetScrollLeft });
+        this.setState({ tabsetMaxScroll });
+        this.setState({ tabsetWidth });
```

Comment:
why too many calls to setState? you can:
```
this.setState({
    tabsetPosition,
    screenWidth,
    ...
});
```
also review this carefully, only put in state reactive things
instead use this.varibleName whenever possible